zeus/optimizer/power_limit.py

```diff
@@ -468,3 +470,150 @@ def _save_profile(self) -> None:
+    """Wrap a `zeus` callback class to be compatible with Hugging Face's Trainer.
+
+    Args:
+        cls (Type[Callback]): The `zeus` callback class to wrap.
+        name (str, optional): The name of the new TrainerCallback class. If None, use `cls.__name__ + "HFWrapper"`.
+
+    """
+    # dynamically import Hugging Face classes, to avoid hard dependency
+    hf_classes = import_hf_classes()
+    if hf_classes is None:
+        raise ImportError(
+            "Hugging Face is not installed. Please install it to use this feature."
+        )
+
+    TrainerCallback, _, _, _, _ = hf_classes  # noqa: N806
+
+    class Wrapper(TrainerCallback):
+        def __init__(self, *args, **kwargs) -> None:
+            self.plo = cls(*args, **kwargs)
+
+        def on_epoch_begin(
+            self,
+            args: TrainingArguments,
+            state: TrainerState,
+            control: TrainerControl,
+            model: PreTrainedModel,
+            **kwargs,
+        ) -> None:
+            self.plo.on_epoch_begin()
+
+        def on_epoch_end(
+            self,
+            args: TrainingArguments,
+            state: TrainerState,
+            control: TrainerControl,
+            model: PreTrainedModel,
+            **kwargs,
+        ) -> None:
+            self.plo.on_epoch_end()
+
+        def on_evaluate(
+            self,
+            args: TrainingArguments,
+            state: TrainerState,
+            control: TrainerControl,
+            model: PreTrainedModel,
+            **kwargs,
+        ) -> None:
+            self.plo.on_evaluate()  # what to set metric to?
+
+        def on_step_begin(
+            self,
+            args: TrainingArguments,
+            state: TrainerState,
+            control: TrainerControl,
+            model: PreTrainedModel,
+            **kwargs,
+        ) -> None:
+            self.plo.on_step_begin()
+
+        def on_step_end(
+            self,
+            args: TrainingArguments,
+            state: TrainerState,
+            control: TrainerControl,
+            model: PreTrainedModel,
+            **kwargs,
+        ) -> None:
+            self.plo.on_step_end()
+
+        def on_train_begin(
+            self,
+            args: TrainingArguments,
+            state: TrainerState,
+            control: TrainerControl,
+            model: PreTrainedModel,
+            **kwargs,
+        ) -> None:
+            self.plo.on_train_begin()
+
+        def on_train_end(
+            self,
+            args: TrainingArguments,
+            state: TrainerState,
+            control: TrainerControl,
+            model: PreTrainedModel,
+            **kwargs,
+        ) -> None:
+            self.plo.on_train_end()
+
+    Wrapper.__name__ = name or (cls.__name__ + "HFWrapper")
+    if cls.__doc__ is not None:
+        Wrapper.__doc__ = "[Wrapped for Hugging Face Trainer Callback] " + cls.__doc__
+
+    return Wrapper
+
+
+HFGPLO = make_hf(GlobalPowerLimitOptimizer)
```

Comment:
Let's fix naming policies to `"HF" + cls.__name__`. The name you set in line 612 should be the same as the name you assign to in this line.